### PR TITLE
enhance: add tool registry revision to version api

### PR DESCRIPTION
**Pull request by njhale**
Add the tool registry (index) and workspace provider revision to the version API.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -42,6 +42,7 @@ set -e
 mkdir -p /run/sshd
 /usr/sbin/sshd -D &
 mkdir -p /data/cache
+export OTTO_SERVER_TOOL_REGISTRY_REVISION=$(cd /otto8-tools && git rev-parse HEAD)
```
Comment (njhale) [inline Dockerfile]: did it this way since Docker's `ENV` doesn't support running commands

File: pkg/api/handlers/version.go [hunk 2]
```diff
@@ -1,10 +1,56 @@
...
+		return "", err
+	}
+
+	for _, toolRef := range toolRefList.Items {
+		if strings.Contains(toolRef.Spec.Reference, toolRegistryURL) && len(toolRef.Spec.Revision) >= 7 {
+			return toolRef.Spec.Revision[:7], nil
+		}
+	}
+
+	return "", nil
```
Comment (njhale) [inline pkg/api/handlers/version.go]: On the fence about dropping this part. Kind of kludgy and not sure how much value it's adding other than for running this locally.

File: Dockerfile (hunk 1 above)
Comment (drpebcak) [reply Dockerfile]: would be nice to pass this in as a build arg instead. we should stop putting things like .git dirs into our docker images.

File: pkg/api/handlers/version.go [hunk 3]
```diff
@@ -1,10 +1,40 @@
...
+			values[parts[0]] = parts[1]
+		}
+	}
+
+	// Populate the versionResponse struct using the map
+	return &versionResponse{
+		Otto:              version.Get().String(),
+		Tools:             strings.TrimSpace(values["Tools"]),
+		WorkspaceProvider: strings.TrimSpace(values["WorkspaceProvider"]),
+	}
```
Comment (njhale) [inline pkg/api/handlers/version.go:39]: I could just do this in an init function so it happens only once. LMK if you want me to change it.

File: pkg/api/handlers/version.go (hunk 2 above)
Comment (njhale) [reply pkg/api/handlers/version.go]: Removed

**Pull request merged**